### PR TITLE
DEV-14034: add generation of private and public keys for okta sso

### PR DESCRIPTION
**Pull request by TomerHasson**

### CODE REVIEW

File: controllers/cnvrgapp_controller.go [hunk 1]
```diff
@@ -408,6 +409,13 @@ func (r *CnvrgAppReconciler) applyManifests(cnvrgApp *mlopsv1.CnvrgApp) error {
 		return err
 	}
 
+	// sso
+	if err := r.ssoState(cnvrgApp); err != nil {
```
Comment (Dimss) [inline controllers/cnvrgapp_controller.go:413]: the PKI for the JWT doesn't related to SSO. 
It's should be always deployed

File: controllers/cnvrgapp_controller.go [hunk 2]
```diff
@@ -567,6 +575,36 @@ func (r *CnvrgAppReconciler) backupsState(app *mlopsv1.CnvrgApp) error {
 	return nil
 }
 
+func (r *CnvrgAppReconciler) ssoState(app *mlopsv1.CnvrgApp) error {
+	// apply sso state
+	if app.Spec.SSO.Pki.Enabled {
```
Comment (Dimss) [inline controllers/cnvrgapp_controller.go:580]: same here, PKI should be enabled outside of SSO

**Review by Dimss — CHANGES_REQUESTED** [review]
@TomerHasson pls apply required chagnes

File: controllers/cnvrgapp_controller.go (hunk 1 above)
Comment (TomerHasson) [reply controllers/cnvrgapp_controller.go:413]: fixed

File: controllers/cnvrgapp_controller.go (hunk 2 above)
Comment (TomerHasson) [reply controllers/cnvrgapp_controller.go:580]: fixed

**Review by Dimss — APPROVED** [review]

**Pull request merged**